### PR TITLE
Add a new test case for backdoor delete feedback question operation #4762

### LINKED ISSUE #4762: Add a new test case for backdoor delete feedback question operation

### PR DESCRIPTION
**Pull request by unyoungwax**
Fixes #4762

### CODE REVIEW

File: src/test/java/teammates/test/cases/testdriver/BackDoorTest.java [hunk 1]
```diff
@@ -512,6 +530,14 @@ private void equalizeIrrelevantData(
         }
     }
 
+    private void equalizeIrrelevantData(
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/testdriver/BackDoorTest.java]: Although it's not done for the other similar methods in this class, consider renaming this method to make it explicit which attributes are getting set.

**Comment by unyoungwax** [comment]
I have renamed the `equalizeIrrelevantData` method to `equalizeId` to show the attribute that is set.

File: src/test/java/teammates/test/cases/testdriver/BackDoorTest.java [hunk 2]
```diff
@@ -512,6 +530,14 @@ private void equalizeIrrelevantData(
         }
     }
 
+    private void equalizeId(
+            FeedbackQuestionAttributes expectedFeedbackQuestion,
+            FeedbackQuestionAttributes actualFeedbackQuestion) {
+
+        // Match the id of the expected FeedbackQuestion because it is not known in advance
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/testdriver/BackDoorTest.java]: Move this comment up to before this method is called? Before line 491

**Comment by kanghj** [comment]
@unyoungwax, remember to comment (or label the pr 'toReview' if you can) whenever your code is ready for review

File: src/test/java/teammates/test/cases/testdriver/BackDoorTest.java [hunk 3]
```diff
@@ -473,7 +484,16 @@ private void verifyPresentInDatastore(AccountAttributes expectedAccount) {
         expectedAccount.studentProfile.modifiedDate = actualAccount.studentProfile.modifiedDate;
         assertEquals(gson.toJson(expectedAccount), gson.toJson(actualAccount));
     }
-    
+
+    private void verifyPresentInDatastore(FeedbackQuestionAttributes expectedQuestion) {
+        String questionJsonString = BackDoor.getFeedbackQuestionAsJson(expectedQuestion.feedbackSessionName, expectedQuestion.courseId, expectedQuestion.questionNumber);
```
Comment (damithc) [inline src/test/java/teammates/test/cases/testdriver/BackDoorTest.java]: line too long

File: src/test/java/teammates/test/cases/testdriver/BackDoorTest.java [hunk 4]
```diff
@@ -512,6 +532,13 @@ private void equalizeIrrelevantData(
         }
     }
 
+    private void equalizeId(
+            FeedbackQuestionAttributes expectedFeedbackQuestion,
+            FeedbackQuestionAttributes actualFeedbackQuestion) {
```
Comment (damithc) [inline src/test/java/teammates/test/cases/testdriver/BackDoorTest.java]: No need for the first line break?

Comment (unyoungwax) [inline src/test/java/teammates/test/cases/testdriver/BackDoorTest.java]: I followed the style of similar methods (e.g. `equalizeIrrelevantData`).

Comment (damithc) [inline src/test/java/teammates/test/cases/testdriver/BackDoorTest.java]: Ok, leave as it is for now. After this is merged, fit it across the board as a separate issue.

**Comment by unyoungwax** [comment]
I have split the line of code that is too long into two instead of renaming the parameter so that the style of similar methods(`verifyPresentInDatastore`) are consistent.

File: src/test/java/teammates/test/cases/testdriver/BackDoorTest.java [hunk 5]
```diff
@@ -473,7 +484,17 @@ private void verifyPresentInDatastore(AccountAttributes expectedAccount) {
         expectedAccount.studentProfile.modifiedDate = actualAccount.studentProfile.modifiedDate;
         assertEquals(gson.toJson(expectedAccount), gson.toJson(actualAccount));
     }
-    
+
+    private void verifyPresentInDatastore(FeedbackQuestionAttributes expectedQuestion) {
+        String questionJsonString = BackDoor.getFeedbackQuestionAsJson(expectedQuestion.feedbackSessionName,
+                                        expectedQuestion.courseId, expectedQuestion.questionNumber);
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/testdriver/BackDoorTest.java]: Just to nitpick, you can try aligning the parameters. 

See
https://docs.google.com/document/pub?id=1iAESIXM0zSxEa5OY7dFURam_SgLiSMhPQtU0drQagrs&embedded=true#h.ukkp0ov105ly
on 
<code>Use line wrapping to improve readability.</code>

**Comment by unyoungwax** [comment]
I have aligned the parameters of `verifyPresentInDatastore` method.

**Pull request merged**